app/src/highlighter/index.ts

```diff
@@ -11,115 +11,243 @@ import {
-extensionMIMEMap.set('.scss', 'text/x-scss')
-extensionMIMEMap.set('.less', 'text/x-less')
-
-import 'codemirror/mode/vue/vue'
-extensionMIMEMap.set('.vue', 'text/x-vue')
-
-import 'codemirror/mode/markdown/markdown'
-extensionMIMEMap.set('.markdown', 'text/x-markdown')
-extensionMIMEMap.set('.md', 'text/x-markdown')
-
-import 'codemirror/mode/yaml/yaml'
-extensionMIMEMap.set('.yaml', 'text/yaml')
-extensionMIMEMap.set('.yml', 'text/yaml')
-
-import 'codemirror/mode/xml/xml'
-extensionMIMEMap.set('.xml', 'text/xml')
-extensionMIMEMap.set('.xaml', 'text/xml')
-extensionMIMEMap.set('.csproj', 'text/xml')
-extensionMIMEMap.set('.fsproj', 'text/xml')
-extensionMIMEMap.set('.vcxproj', 'text/xml')
-extensionMIMEMap.set('.vbproj', 'text/xml')
-extensionMIMEMap.set('.svg', 'text/xml')
-
-import 'codemirror/mode/clike/clike'
-extensionMIMEMap.set('.m', 'text/x-objectivec')
-extensionMIMEMap.set('.scala', 'text/x-scala')
-extensionMIMEMap.set('.sc', 'text/x-scala')
-extensionMIMEMap.set('.cs', 'text/x-csharp')
-extensionMIMEMap.set('.java', 'text/x-java')
-extensionMIMEMap.set('.c', 'text/x-c')
-extensionMIMEMap.set('.h', 'text/x-c')
-extensionMIMEMap.set('.cpp', 'text/x-c++src')
-extensionMIMEMap.set('.hpp', 'text/x-c++src')
-extensionMIMEMap.set('.kt', 'text/x-kotlin')
-
-import 'codemirror/mode/mllike/mllike'
-extensionMIMEMap.set('.ml', 'text/x-ocaml')
-extensionMIMEMap.set('.fs', 'text/x-fsharp')
-extensionMIMEMap.set('.fsx', 'text/x-fsharp')
-extensionMIMEMap.set('.fsi', 'text/x-fsharp')
-
-import 'codemirror/mode/swift/swift'
-extensionMIMEMap.set('.swift', 'text/x-swift')
-
-import 'codemirror/mode/shell/shell'
-extensionMIMEMap.set('.sh', 'text/x-sh')
-
-import 'codemirror/mode/sql/sql'
-extensionMIMEMap.set('.sql', 'text/x-sql')
-
-import 'codemirror/mode/cypher/cypher'
-extensionMIMEMap.set('.cql', 'application/x-cypher-query')
-
-import 'codemirror/mode/go/go'
-extensionMIMEMap.set('.go', 'text/x-go')
-
-import 'codemirror/mode/perl/perl'
-extensionMIMEMap.set('.pl', 'text/x-perl')
-
-import 'codemirror/mode/php/php'
-extensionMIMEMap.set('.php', 'application/x-httpd-php')
-
-import 'codemirror/mode/python/python'
-extensionMIMEMap.set('.py', 'text/x-python')
-
-import 'codemirror/mode/ruby/ruby'
-extensionMIMEMap.set('.rb', 'text/x-ruby')
-
-import 'codemirror/mode/clojure/clojure'
-extensionMIMEMap.set('.clj', 'text/x-clojure')
-extensionMIMEMap.set('.cljc', 'text/x-clojure')
-extensionMIMEMap.set('.cljs', 'text/x-clojure')
-extensionMIMEMap.set('.edn', 'text/x-clojure')
+/**
+ * A mode definition object is used to map a certain file
+ * extension to a mode loader (see the documentation for
+ * the require property).
+ */
+interface IModeDefinition {
+  /**
+   * A function that, when called, will attempt to asynchronously
+   * load the required modules for a particular mode. This function
+   * is idempotent and can be called multiple times with no adverse
+   * effect.
+   */
+  readonly require: () => Promise<void>
```

Comment:
How do you feel about calling this `import` or `install` to encourage people.m not to use `require`?

Reply:
I’m kinda liking `install` since the return value is ignored and the function has a side effect, which is what installation typically means.

Reply:
I have no feelings whatsoever on what we name this but `import` is a reserved keyword which is why I didn't go with that to begin with. So `install` it is.